syncthing-bep/src/main/kotlin/net/syncthing/java/bep/BlockPuller.kt

```diff
@@ -84,22 +88,27 @@ object BlockPuller {
                             // is handled differently so that the timeout is ignored.
                             // Due to that, it's converted to an IOException.
 
-                            throw IOException("timeout during requesting block")
+                            val exception = IOException("Time limit exceeded while requesting block.")
+                            LOGGER.atInfo().withThrowable(exception).log("Timeout set for {} seconds.", box(timeoutInMillis/1000))
+                            throw exception
                         }
                     }
 
             if (response.code != BlockExchangeProtos.ErrorCode.NO_ERROR) {
                 // the server does not have/ want to provide this file -> don't ask him again
                 connectionHelper.disableConnection(connectionActorWrapper)
-
-                throw IOException("received error response ${response.code}")
+                val exception = IOException("Server does not have or is not able to provide the requested file.")
+                LOGGER.atDebug().withThrowable(exception).log("Received following error response from server: {}.", response.code)
```

Comment:
`box()`? Don't include the exception. error level. log & exception should have similar information in their message

Reply:
It is not a boxable argument as far as I understand.